ui/chat/src/discussion.ts

```diff
@@ -84,11 +84,18 @@ function renderInput(ctrl: Ctrl): VNode | undefined {
 let mouchListener: EventListener;
 
 const setupHooks = (ctrl: Ctrl, chatEl: HTMLElement) => {
-  chatEl.addEventListener('keypress',
+  chatEl.addEventListener('keydown',
     (e: KeyboardEvent) => setTimeout(() => {
       const el = e.target as HTMLInputElement,
         txt = el.value,
         pub = ctrl.opts.public;
+      if(sessionStorage.getItem("txt") != null && sessionStorage.getItem("txt") != "") {
```

Comment:
You definitely don't want to take the `"txt"` key for yourself. Who knows what other code could be using it? Need a more specific key.

For example, if you switch tournaments, should the chat message be conserved? Maybe you need some kind of id in there?